finetuner/tuner/dataset/samplers.py

```diff
@@ -7,6 +7,46 @@
 from finetuner.tuner.dataset.base import BaseSampler
 
 
+class InstanceSampler(BaseSampler):
+    """A sampler that includes multiple copies of an instance in the batch.
+
+    Used for self-supervised learning, where positive samples for an anchor are just
+    (differently) augmented copies of itself.
+    """
+
+    def __init__(self, num_instances: int, batch_size: int, repeat_instance: int = 2):
+        """Construct the instance sampler.
+
+        :param num_instances: Number of instances in the dataset
+        :param batch_size: How many items to include in a batch
+        :param repeat_instance: How many times to repeat each instance in a batch
+        """
+
+        if batch_size <= 0:
+            raise ValueError('batch_size must be a positive integer')
+
+        if batch_size % repeat_instance != 0:
```

Comment:
we might also don't want to set `repeat_instance` to 1? otherwise SSL learns nothing

Reply:
Good idea